### PR TITLE
Adding VideoClassificationDirectoryTree, HMDB51, UCF101, BDD100k, and COCO w/ segmentations

### PR DESCRIPTION
**Pull request by brimoor**
Adds a `VideoClassificationDirectoryTree` dataset format that can be used to load video classification datasets stored on disk as directory trees.

Leverages this new type to add HMDB51 and UCF101, two video classification datasets, to the dataset zoo.

Also adds BDD100K to the dataset zoo, and adds new versions of COCO 2014/2017 that contain instance segmentations to the dataset zoo (the existing `coco-2014/coco-2017` datasets only load the detections).

Treating with and without segmentations as separate COCO datasets makes sense, at least for now, because instance segmentations take quite a bit of space, and performance is a key concern.

In order to add support for video classification datasets, some enhancements needed to be made to the `LabeledVideoDatasetImporter/Exporter` interface, so that importers/exporters for labeled video datasets can specify both sample and frame-level labels that they support.

Lastly, I was finally able to completely get rid of the `fiftyone.core.labels.ImageLabels` type, since the best practice has emerged to always "expand" multitask types like that into individual fields.

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -17,7 +17,17 @@ Tests do exist, but their coverage generally needs improvement...
 
 ## Running a test
 
-To run a test, simply run `python <name-of-file>.py`
+To run all unit tests, run:
+
+```shell
+bash unittests/run_all.bash
```
Comment (lethosor) [inline tests/README.md:23]: I would heavily encourage `pytest unittests/` over this - we're already using this in CI and it has been working well.

File: tests/README.md [hunk 2]
```diff
@@ -17,7 +17,17 @@ Tests do exist, but their coverage generally needs improvement...
...
+```
+
+To run a specific test, do:
+
+```shell
+python unittests/<file>.py
```
Comment (lethosor) [inline tests/README.md:29]: Similarly, `pytest unittests/<file>.py`

File: tests/README.md (hunk 1 above)
Comment (brimoor) [reply tests/README.md]: Thanks I didn't know that was possible when not all tests are using `pytest`

**Comment by benjaminpkane** [comment]
Are you seeing `MediaTypeError`s for `VideoClassificationDirectoryTree`? I can't recall if it was decided that we will support all label types at the sample-level for video datasets, or just classification types? In either case, I think `fiftyone.core.media.validate_field_against_media_type()`'s usage or definition needs to updated (or removed).

`fiftyone.core.utils.annotations` has `fol.ImageLabel` usage that will need to updated/removed, as well, correct?

I am generally opposed to supporting sample-level labels for video datasets that don't make sense (`Detections`, etc.). Supporting that in player51 won't be natural with a fair amount of changes, I think.

**Comment by benjaminpkane** [comment]
> `fiftyone.core.utils.annotations` has `fol.ImageLabel` usage that will need to updated/removed, as well, correct?

My mistake, I was confusing `ImageLabels` and `ImageLabel`.

**Comment by brimoor** [comment]
> Are you seeing `MediaTypeError`s for `VideoClassificationDirectoryTree`? I can't recall if it was decided that we will support all label types at the sample-level for video datasets, or just classification types? In either case, I think `fiftyone.core.media.validate_field_against_media_type()`'s usage or definition needs to updated (or removed).

@benjaminpkane Ah yes I'm seeing that too. I wonder how I was *not* seeing that before.... Hmmm. Now that its perfectly legit to have `Classification` at the sample-level for videos, I think this util method is deprecated. Okay with removing it? https://github.com/voxel51/fiftyone/pull/612/commits/e5f9da4561f0df67003b9b9fc3408aadd069e56d.

I suppose it generally doesn't make any sense to put a `Detection` at the video-level. But, rather than preventing it from happening in the library, if we're not going to support visualizing it, then I'd rather just make it appear as unsupported in the App.

**Comment by benjaminpkane** [comment]
Sounds good, agree with all of the above.

**Review by benjaminpkane — APPROVED** [review]
LGTM, assuming we are okay with Classifications on video samples not being acknowledged yet in the App.

Nice to see the `ImageLabels` cleanup!

**Pull request merged**